pkg/venafi/cloud/connector.go

```diff
@@ -753,6 +760,150 @@ func (c *Connector) SearchCertificate(zone string, cn string, sans *certificate.
+		return nil, fmt.Errorf("only should keystoreId OR keystoreName and providerName can be provided")
+	}
+	if keystoreId == nil {
+		if keystoreName == nil {
+			return nil, fmt.Errorf("missing providerName")
+		}
+		if providerName == nil {
+			return nil, fmt.Errorf("missing keystoreName")
+		}
+		return nil, fmt.Errorf("missing to provide any of keystoreId or keystoreName and providerName")
+	}
+
+	ctx := context.Background()
+
+	// Is certificate generated by VCP?
+	err = c.validateIfCertIsVCPGeneratedByID(certificateId)
+	if err != nil {
+		return nil, err
+	}
+
+	if keystoreId == nil && (keystoreName != nil && providerName != nil) {
+		// TODO: Add logic to get keystore ID from Keystore's Name and its Provider Name once filters are ready
+		// here should feed up "keystoreId" var for next validation
+		if keystoreId == nil {
+			return nil, fmt.Errorf("could not find Keystore ID for Keystore Name %s and Cloud Provider Name %s", *keystoreName, *providerName)
+		}
+	}
+
+	graphqlClient := c.getGraphqlClient()
+
+	// Getting Keystore to find type
+	data, err := cloudkeystores.GetCloudKeystoresByKeystoreId(ctx, graphqlClient, keystoreId)
+	if err != nil {
+		return nil, err
+	}
+
+	keystoreIdString := *keystoreId
+
+	if len(data.CloudKeystores.Nodes) == 0 {
+		return nil, fmt.Errorf("could not find keystore with ID: %s", keystoreIdString)
+	}
+
+	if len(data.CloudKeystores.Nodes) > 1 {
+		return nil, fmt.Errorf("found more than one keystore for ID: %s", keystoreIdString)
+	}
+
+	// grabbing found keystore for later
+	cloudKeystore := data.CloudKeystores.Nodes[0]
+
+	wsClientId := uuid.New()
+
+	wsClientIdString := wsClientId.String()
+
+	wsConn, err := websocket.Subscribe(c.apiKey, c.baseURL, wsClientIdString)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = cloudkeystores.ProvisionCertificate(ctx, graphqlClient, certificateId, keystoreIdString, wsClientIdString, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	ar, err := websocket.ReadResponse(wsConn)
+	if err != nil {
+		return nil, err
+	}
+
+	respMap := ar.Data.Result
+	val := CloudKeystoreProvisioningResult{}
+	valJs, err := json.Marshal(respMap)
+	if err != nil {
+		return nil, fmt.Errorf("unable to encode response data! Error: %s", err.Error())
+	}
+	err = json.Unmarshal(valJs, &val)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse response data! Error: %s", err.Error())
+	}
+	if val.Error != nil {
+		return nil, fmt.Errorf("unable to provision certificate! Error: %s", val.Error)
+	}
+
+	if val.CloudProviderCertificateId == "" {
+		return nil, fmt.Errorf("provisioning is not successful, certificate ID from response is empty")
+	}
+
+	log.Printf("Successfully provisioned certificate with CertificateId: %s", val.CloudProviderCertificateId)
+
+	var cloudMetadata CloudProvisioningMetadata
+	switch cloudKeystore.Type {
+	case cloudkeystores.CloudKeystoreTypeAcm:
+		cloudMetadata.awsMetadata.result = val
+		cloudMetadata.GetAwsMetadata().GetARN()
```

Comment:
What do all these GetXXX calls do (since their return is not assigned to anything)?

Reply:
Sorry, yeah, it's a left over. I'm removing in next commit

Reply:
Fixed by latest commit